cpp/tests/bitmask/bitmask_tests.cu

```diff
@@ -62,149 +62,323 @@ struct CountBitmaskTest : public cudf::test::BaseFixture {};
+  auto counts = cudf::segmented_count_unset_bits(mask.data().get(), indices);
+  EXPECT_EQ(2, static_cast<cudf::size_type>(counts.size()));
+  EXPECT_EQ(30, counts[0]);
+  EXPECT_EQ(10, counts[1]);
 }
 
 TEST_F(CountUnsetBitsTest, SingleWordSubset2) {
   thrust::device_vector<cudf::bitmask_type> mask(1, cudf::bitmask_type{0});
   EXPECT_EQ(28, cudf::count_unset_bits(mask.data().get(), 2, 30));
+
+  std::vector<cudf::size_type> indices = {4, 16, 2, 30};
+  auto counts = cudf::segmented_count_unset_bits(mask.data().get(), indices);
+  EXPECT_EQ(2, static_cast<cudf::size_type>(counts.size()));
+  EXPECT_EQ(12, counts[0]);
+  EXPECT_EQ(28, counts[1]);
 }
 
 TEST_F(CountUnsetBitsTest, MultipleWordsAllBits) {
   thrust::device_vector<cudf::bitmask_type> mask(10, cudf::bitmask_type{0});
   EXPECT_EQ(320, cudf::count_unset_bits(mask.data().get(), 0, 320));
+
+  std::vector<cudf::size_type> indices = {0, 320, 0, 320};
+  auto counts = cudf::segmented_count_unset_bits(mask.data().get(), indices);
+  EXPECT_EQ(2, static_cast<cudf::size_type>(counts.size()));
+  EXPECT_EQ(320, counts[0]);
+  EXPECT_EQ(320, counts[1]);
 }
 
 TEST_F(CountUnsetBitsTest, MultipleWordsSubsetWordBoundary) {
   thrust::device_vector<cudf::bitmask_type> mask(10, cudf::bitmask_type{0});
   EXPECT_EQ(256, cudf::count_unset_bits(mask.data().get(), 32, 288));
+
+  std::vector<cudf::size_type> indices = {32, 192, 32, 288};
+  auto counts = cudf::segmented_count_unset_bits(mask.data().get(), indices);
+  EXPECT_EQ(2, static_cast<cudf::size_type>(counts.size()));
+  EXPECT_EQ(160, counts[0]);
+  EXPECT_EQ(256, counts[1]);
 }
 
 TEST_F(CountUnsetBitsTest, MultipleWordsSplitWordBoundary) {
   thrust::device_vector<cudf::bitmask_type> mask(10, cudf::bitmask_type{0});
   EXPECT_EQ(2, cudf::count_unset_bits(mask.data().get(), 31, 33));
+
+  std::vector<cudf::size_type> indices = {31, 33, 60, 67};
+  auto counts = cudf::segmented_count_unset_bits(mask.data().get(), indices);
+  EXPECT_EQ(2, static_cast<cudf::size_type>(counts.size()));
+  EXPECT_EQ(2, counts[0]);
+  EXPECT_EQ(7, counts[1]);
 }
 
 TEST_F(CountUnsetBitsTest, MultipleWordsSubset) {
   thrust::device_vector<cudf::bitmask_type> mask(10, cudf::bitmask_type{0});
   EXPECT_EQ(226, cudf::count_unset_bits(mask.data().get(), 67, 293));
+
+  std::vector<cudf::size_type> indices = {67, 293, 37, 319};
+  auto counts = cudf::segmented_count_unset_bits(mask.data().get(), indices);
+  EXPECT_EQ(2, static_cast<cudf::size_type>(counts.size()));
+  EXPECT_EQ(226, counts[0]);
+  EXPECT_EQ(282, counts[1]);
 }
 
 TEST_F(CountUnsetBitsTest, MultipleWordsSingleBit) {
   thrust::device_vector<cudf::bitmask_type> mask(10, cudf::bitmask_type{0});
   EXPECT_EQ(1, cudf::count_unset_bits(mask.data().get(), 67, 68));
+
+  std::vector<cudf::size_type> indices = {67, 68, 31, 32, 192, 193};
+  auto counts = cudf::segmented_count_unset_bits(mask.data().get(), indices);
+  EXPECT_EQ(3, static_cast<cudf::size_type>(counts.size()));
```

Comment:
You can check the entire vector in a single line, something like
ASSERT_THAT(counts, ::testing::ContainerEq({1,1,1}));